tests/spanner_dbapi/test_checksum.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Google LLC
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bsd
+
+import unittest
+
+from google.cloud.spanner_dbapi.checksum import (
+    _compare_checksums,
+    ResultsChecksum,
+)
+
+
+class Test_compare_checksums(unittest.TestCase):
+    def test_no_original_checksum(self):
+        self.assertIsNone(_compare_checksums(None, ResultsChecksum()))
+
+    def test_equal(self):
+        original = ResultsChecksum()
+        original.consume_result(5)
+
+        retried = ResultsChecksum()
+        retried.consume_result(5)
+
+        self.assertIsNone(_compare_checksums(original, retried))
+
+    def test_less_results(self):
+        original = ResultsChecksum()
+        original.consume_result(5)
+
+        retried = ResultsChecksum()
+
+        self.assertIsNone(_compare_checksums(original, retried))
```

Comment:
I think this is a little bit of a misunderstanding. Two result sets that have consumed different number of rows, should never have equal checksums.

Reply:
I've made it like this: checksum tracks the number of consumed results internally. When one is comparing two checksums, the first thing that happens is their length comparement. If they have equal length and different checksums, then the check will raise `Aborted` exception. In all other cases it'll do nothing and will return `None`.